src/main/java/com/nirmata/workflow/queue/zookeeper/SimpleQueue.java

```diff
@@ -200,6 +200,22 @@ public void debugValidateClosed()
         Preconditions.checkState(executorService.isTerminated());
     }
 
+    @Override
+    public void closeGraceFully(long timeOut, TimeUnit unit) {
+        if ( started.compareAndSet(true, false) )
+        {
+            executorService.shutdown();
```

Comment:
Guava has `MoreExecutors.shutdownAndAwaitTermination`. Consider using it instead.